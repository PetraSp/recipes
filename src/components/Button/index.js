import Styled from 'styled-components';

const Button = Styled.button`
  display: inline-block;
  border: 0;
  outline: 0;
  margin-top: 16px;
  padding: 14px 16px;
  font-size: 14px;
  font-weight: 700;
  border-radius: 4px;
  background-color: #2eb0a8;
  color: #fff;
  
  :hover {
    cursor: pointer;
    text-decoration: none;
    color: #fff;
    box-shadow: 0 3px 5px rgba(0,0,0,.12);
  }
}
  `;

export default Button;
