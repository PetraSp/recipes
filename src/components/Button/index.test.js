import React from 'react';
import renderer from 'react-test-renderer';
import Button from './index';
import 'jest-styled-components';

describe('Button', () => {
  it('should render', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('background-color', '#2eb0a8');
    expect(tree).toHaveStyleRule('color', '#fff');
    expect(tree).toHaveStyleRule('border-radius', '4px');
  });
});
