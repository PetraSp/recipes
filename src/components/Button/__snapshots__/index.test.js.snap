// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
.c0 {
  display: inline-block;
  border: 0;
  outline: 0;
  margin-top: 16px;
  padding: 14px 16px;
  font-size: 14px;
  font-weight: 700;
  border-radius: 4px;
  background-color: #2eb0a8;
  color: #fff;
}

.c0:hover {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
  box-shadow: 0 3px 5px rgba(0,0,0,.12);
}

<button
  className="c0"
/>
`;
